src/main/scala/io/iohk/praos/domain/Block.scala

```diff
@@ -0,0 +1,21 @@
+package io.iohk.praos.domain
+
+import io.iohk.praos.crypto.{Hasher, Seed, Signature, VrfProof, combineSeeds}
+
+
+case class Block(
+  state       : Option[Hasher#Digest],
+  slotNumber  : SlotNumber,
+  data        : List[Transaction],
+  proof       : VrfProof,
+  nonce       : (Seed, VrfProof),
+  signature   : Signature) {
+
+  def apply(genesisBlock: GenesisBlock): GenesisBlock = {
```

Comment:
Maybe we should rename `GenesisBlock` to `VirtualGenesis` or something like that to differentiate it from the block 0?

Reply:
The Praos paper explicitly defines it as "genesis **block**" and differentiates it from a block in a block chain, whereas the Praos formalization calls it merely "genesis". I agree with you in that it's not properly a "block", but personally I don't like the adjective "virtual" (what would a "real genesis" be then?), so I would rather stick to the Praos formalization and call it "genesis".

You're right in that the result of applying a `Block` to a `GenesisBlock` is normally an "intermediate genesis", but I think it's not worth it to define another type for those intermediate results, which would turn out to have exactly the same structure.

Reply:
Discussed with Nico, we will stick to the Praos paper.

Reply:
Minor detail: Maybe we could add a comment on the `GenesisBlock` class regarding this uses of it?

Reply:
Sounds good, will do.

Reply:
Resolved.